wordcount_and_median.py

```diff
@@ -43,46 +42,32 @@ def __cleaner_lister(incoming_string):
     outgoing_list = incoming_string.split()
     return outgoing_list
 
-     
 
 def count_writer(output_folder, wc_out_file, fulldict_cw):
-    with open(output_folder+'/'+wc_out_file, 'w') as d:
-        for item in sorted( fulldict_cw.keys() ):
-            out_line = str(item)+'\t'+str( fulldict_cw[item] )+'\n'
-            d.write(out_line)
-        d.closed
+    with open(os.path.join(output_folder, wc_out_file), 'w') as d:
+        writer = csv.writer(d, delimiter='\t')
+        for word, count in sorted(fulldict_cw.items()):
+            writer.writerow([word, str(count)])
 
 
 def median_writer(output_folder, med_out_file, linelengths_mw):
-    sorted_list = [];
-    with open(output_folder+'/'+med_out_file, 'w') as m:
-        for item in range( len(linelengths_mw) ):
-            bisect.insort(sorted_list, linelengths_mw[item]);
-            m.write('{0:.1f}\n'.format( __presorted_median(sorted_list) ) )
-        m.closed
+    sorted_list = []
+    with open(os.path.join(output_folder, med_out_file), 'w') as m:
+        for linelength in linelengths_mw:
+            bisect.insort(sorted_list, linelength)
+            m.write('{0:.1f}\n'.format( _presorted_median(sorted_list) ) )
 
 
-def __presorted_median(data):
+def _presorted_median(data):
     n = len(data)
     if n == 0:
```

Comment:
Minor style changes:

1) The Python style guide says you should always put spaces around math operators such as ``%`` and ``/`` and ``//``.

2) There's no agreed-upon universal preference or whether to use single quotes or double quotes for strings, but it's generally agreed that we should be consistent in whichever we choose.  Since you used single quotes above, I changed it to use single quotes here.